scripts/set_151_officer.py

```diff
@@ -0,0 +1,130 @@
+"""
+==============
+Pre-requisites
+=============
+Set the below environmnetal variable
+export FLASK_ENV=development
+export ACCOUNT_STORE_API_HOST=http://localhost:3003
+
+==============
+To Run
+=============
+# Create or update an account with the role "OFFICER_151"
+>> python -m scripts.set_151_officer <email_address1> <email_address2> ...
+"""
+# import json
+import json
+import re
+import sys
+from typing import Optional
+
+import requests
+from config import Config
+
+# Regular expression for validating an Email
+EMAIL_REGEX = r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)"
+
+
+def is_valid_email(email):
+    return re.match(EMAIL_REGEX, email) is not None
+
+
+def get_account(email) -> Optional[dict]:
+    try:
+        response = requests.get(
+            f"{Config.ACCOUNT_STORE_API_HOST}{Config.ACCOUNTS_ENDPOINT}",
+            headers={"Content-Type": "application/json"},
+            params={"email_address": email},
+        )
+
+        response.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        if e.response.status_code == 404:
+            return None
+
+        raise Exception(f"An error occurred: {e.response.text}")
+
+    return response.json()
+
+
+def create_account(email) -> dict:
+    try:
+        response = requests.post(
+            f"{Config.ACCOUNT_STORE_API_HOST}{Config.ACCOUNTS_ENDPOINT}",
+            headers={"Content-Type": "application/json"},
+            data=json.dumps({"email_address": email}),
+        )
+
+        response.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        raise Exception(f"An error occurred: {e.response.text}")
+
+    return response.json()
+
+
+def update_account(
+    account_id: str,
+    email_address: str,
+    azure_ad_subject_id: str,
+    full_name: str,
+    roles: list[str],
+) -> dict:
+    url = Config.ACCOUNT_STORE_API_HOST + Config.ACCOUNT_ENDPOINT.format(account_id=account_id)
+
+    data = {
+        "email_address": email_address,
+        "azure_ad_subject_id": azure_ad_subject_id,
+        "full_name": full_name,
+        "roles": roles,
+    }
+
+    try:
+        response = requests.put(
+            url,
+            headers={"Content-Type": "application/json"},
+            data=json.dumps({k: v for k, v in data.items() if v is not None}),
+        )
+
+        response.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        raise Exception(f"An error occurred: {e.response.text}")
+
+    return response.json()
+
+
+if __name__ == "__main__":
+    if len(sys.argv) < 2:
+        print("Usage: python -m scripts.set_151_officer <email_address1> <email_address2> ...")
+        exit(1)
+
+    emails = sys.argv[1:]
+    for email in emails:
```

Comment:
If any of these loops raise an error, the entire script dies. Should we catch errors for each loop and log them, then move onto the next to try it, rather than failing the entire script?

I suppose the entire script is idempotent and can just be re-run in full, so it probably doesn't matter much 🤔